pandas/core/resample.py

```diff
@@ -2539,17 +2540,28 @@ def _set_grouper(
         return obj, ax, indexer
 
 
+@overload
 def _take_new_index(
-    obj: NDFrameT,
+    obj: DataFrame, indexer: npt.NDArray[np.intp], new_index: Index
+) -> DataFrame:
+    ...
+
+
+@overload
+def _take_new_index(
+    obj: Series, indexer: npt.NDArray[np.intp], new_index: Index
+) -> Series:
+    ...
+
+
+def _take_new_index(
+    obj: DataFrame | Series,
     indexer: npt.NDArray[np.intp],
     new_index: Index,
-) -> NDFrameT:
+) -> DataFrame | Series:
```

Comment:
Had to switch to the overloads as Series._constructor returns type[Series] and not type[Self]